src/app/models/submission.server.model.ts

```diff
@@ -175,6 +183,113 @@ EncryptSubmissionSchema.methods.getWebhookView = function (
   }
 }
 
+EncryptSubmissionSchema.statics.findSingleMetadata = function (
+  this: IEncryptSubmissionModel,
+  formId: string,
+  submissionId: string,
+): Promise<SubmissionMetadata | null> {
+  return (
+    this.findOne(
+      {
+        form: formId,
+        _id: submissionId,
+        submissionType: SubmissionType.Encrypt,
+      },
+      { created: 1 },
+    )
+      // Reading from primary to avoid any contention issues with bulk queries
+      // on secondary servers.
+      .read('primary')
+      .then((result) => {
+        if (!result) {
+          return null
+        }
+
+        // Build submissionMetadata object.
+        const metadata: SubmissionMetadata = {
+          number: 1,
+          refNo: result._id,
+          submissionTime: moment(result.created)
+            .tz('Asia/Singapore')
+            .format('Do MMM YYYY, h:mm:ss a'),
+        }
+
+        return metadata
+      })
+  )
+}
+
+EncryptSubmissionSchema.statics.findAllMetadataByFormId = function (
+  this: IEncryptSubmissionModel,
+  formId: string,
+  {
+    page = 1,
+    pageSize = 10,
+  }: {
+    page?: number
+    pageSize?: number
+  } = {},
+): Promise<{
+  metadata: SubmissionMetadata[]
+  count: number
+}> {
+  const numToSkip = (page - 1) * pageSize
+
+  type MetadataAggregateResult = {
+    pageResults: Pick<ISubmissionSchema, '_id' | 'created'>[]
+    allResults: FixedLengthArray<{ count: number }, 1> | []
+  }
```

Comment:
Define the type outside the function

Reply:
This type is exclusively used in this function, I'm scared types gets too polluted if I move it out

Reply:
you don't have to export it though, and otherwise, the type gets re-declared on every function call